src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.targets

```diff
@@ -128,6 +131,72 @@
     <_SharedRuntimeAssemblies Include="$(_SharedRuntimeBuildPath)$(AndroidFrameworkVersion)\OpenTK.dll" />
     <_SharedRuntimeAssemblies Include="$(_SharedRuntimeBuildPath)$(AndroidFrameworkVersion)\OpenTK-1.0.dll" />
   </ItemGroup>
+  <Target Name="_PrepareNDKToolsItems">
+    <PropertyGroup>
+      <_PrebuiltToolchainDir Condition=" '$(HostOS)' == 'Linux' ">linux-x86_64</_PrebuiltToolchainDir>
+      <_PrebuiltToolchainDir Condition=" '$(HostOS)' == 'Darwin' ">darwin-x86_64</_PrebuiltToolchainDir>
+      <_PrebuiltToolchainDir Condition=" '$(HostOS)' == 'Windows' And '$(HostBits)' == '64' ">windows-x86_64</_PrebuiltToolchainDir>
+      <_PrebuiltToolchainDir Condition=" '$(HostOS)' == 'Windows' And '$(HostBits)' == '32' ">windows</_PrebuiltToolchainDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <_ToolchainPrefix Include="aarch64-linux-android">
+        <ToolPrefix>aarch64-linux-android</ToolPrefix>
+      </_ToolchainPrefix>
+      <_ToolchainPrefix Include="arm-linux-androideabi">
+        <ToolPrefix>arm-linux-androideabi</ToolPrefix>
+      </_ToolchainPrefix>
+      <_ToolchainPrefix Include="x86">
+        <ToolPrefix>i686-linux-android</ToolPrefix>
+      </_ToolchainPrefix>
+      <_ToolchainPrefix Include="x86_64">
+        <ToolPrefix>x86_64-linux-android</ToolPrefix>
+      </_ToolchainPrefix>
+    </ItemGroup>
+
+    <Which
+        HostOS="$(HostOS)"
+        HostOSName="$(HostOsName)"
+        Program="$(AndroidNdkFullPath)/toolchains/%(_ToolchainPrefix.Identity)-4.9/prebuilt/$(_PrebuiltToolchainDir)/bin/%(_ToolchainPrefix.ToolPrefix)-as"
+        Required="True">
+      <Output TaskParameter="Location" PropertyName="_NDKToolLocation%(_ToolchainPrefix.Identity)" />
+    </Which>
+
+    <PropertyGroup>
+      <_NDKToolFileName>$([System.IO.Path]::GetFileName ('$(_NDKToolLocation)'))</_NDKToolFileName>
+      <_WinAsmDestDir>$(XAInstallPrefix)xbuild\Xamarin\Android</_WinAsmDestDir>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_NDKToolSource Include="$(_NDKToolLocation%(_ToolchainPrefix.Identity))" />
+      <_NDKToolDestination Include="$(XAInstallPrefix)xbuild\Xamarin\Android\$(HostOS)\$([System.IO.Path]::GetFileName ('$(_NDKToolLocation%(_ToolchainPrefix.Identity))'))" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_FetchWindowsAssemblers"
+          Condition=" '$(HostOS)' != 'Windows' "
```

Comment:
Indentation: attributes don't need to line up, they just need to be indented *more* than the child elements (to make it easier while skimming to differentiate child elements from attributes):

```xml
  <Target Name="_FetchWindowsAssemblers"
      Condition=" '$(HostOS)' != 'Windows' "
      ...
    <MakeDir
...
```